### PR TITLE
Vendor gardener@1.57.0 and Use extension specific error codes.

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area open-source
/kind enhancement
/platform aws

**What this PR does / why we need it**:
Vendor gardener v1.57.0.
This PR removes extension dependency on g/g to extract error codes from errors and uses regex defined in the extension repo. Later extensions can be adapted to map provider-specific error codes to gardener error codes.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/5444

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependency is updated:
- github.com/gardener/gardener: v1.56.0 -> v1.57.0
- sigs.k8s.io/controller-tools: v0.9.2 -> v0.10.0
```

### CODE REVIEW

**Comment by acumino** [comment]
> Testrun: [e2e-mwcj8](https://tm.gardener.cloud/testrun/default/e2e-mwcj8) Workflow: e2e-mwcj8-wf Phase: Failed
> 
> +---------------------+---------------------+--------+----------+
> |        NAME         |        STEP         | PHASE  | DURATION |
> +---------------------+---------------------+--------+----------+
> | bastion-test        | bastion-test        | Failed | 2m1s     |
> | dnsrecord-test      | dnsrecord-test      | Failed | 2m1s     |
> | infrastructure-test | infrastructure-test | Failed | 2m0s     |
> +---------------------+---------------------+--------+----------+

TM tests are failing because of mismatched go version. go version needs to be updated to 1.19.
/cc @shafeeqes  @kon-angelo

File: pkg/controller/healthcheck/add.go [hunk 1]
```diff
@@ -93,16 +98,19 @@ func RegisterHealthChecks(mgr manager.Manager, opts healthcheck.DefaultAddArgs)
...
+				ConditionType:      string(gardencorev1beta1.ShootSystemComponentsHealthy),
+				HealthCheck:        general.CheckManagedResource(genericcontrolplaneactuator.StorageClassesChartResourceName),
+				ErrorCodeCheckFunc: errorCodeCheckFunc,
 			},
 			{
-				ConditionType: string(gardencorev1beta1.ShootSystemComponentsHealthy),
-				HealthCheck:   general.CheckManagedResource(genericcontrolplaneactuator.ShootWebhooksResourceName),
+				ConditionType:      string(gardencorev1beta1.ShootSystemComponentsHealthy),
+				HealthCheck:        general.CheckManagedResource(genericcontrolplaneactuator.ShootWebhooksResourceName),
+				ErrorCodeCheckFunc: errorCodeCheckFunc,
```
Comment (rfranzke) [inline pkg/controller/healthcheck/add.go:113]: Which kind of provider-specific errors do you expect here?

File: pkg/controller/healthcheck/add.go [hunk 2]
```diff
@@ -137,16 +145,18 @@ func RegisterHealthChecks(mgr manager.Manager, opts healthcheck.DefaultAddArgs)
 		nil,
 		[]healthcheck.ConditionTypeToHealthCheck{
 			{
-				ConditionType: string(gardencorev1beta1.ShootSystemComponentsHealthy),
-				HealthCheck:   general.CheckManagedResource(genericworkeractuator.McmShootResourceName),
+				ConditionType:      string(gardencorev1beta1.ShootSystemComponentsHealthy),
+				HealthCheck:        general.CheckManagedResource(genericworkeractuator.McmShootResourceName),
+				ErrorCodeCheckFunc: errorCodeCheckFunc,
```
Comment (rfranzke) [inline pkg/controller/healthcheck/add.go:150]: Which kind of provider-specific errors do you expect here?

File: pkg/apis/aws/helper/errors.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		}
+	}
+
+	// compute error code list based on code string set
+	var out []gardencorev1beta1.ErrorCode
+	for _, c := range codes.List() {
+		out = append(out, gardencorev1beta1.ErrorCode(c))
+	}
+	return out
+}
```
[the remark this answers is not in the document]
Comment (acumino) [reply pkg/apis/aws/helper/errors.go:101]: If we keep it in the gardener then we will make extension again dependent on gardener, in general one can think they can use SDK-based error code once Gardner removes `DetermineErrorCodes` and similar function.

[the remark this answers is not in the document]
Comment (acumino) [reply pkg/apis/aws/helper/errors.go:101]: In that case, we need to wait for one more release of gardener/gardener.

**Pull request closed without merging**